sros2/sros2/keystore/_keystore.py

```diff
@@ -61,61 +59,53 @@ def create_keystore(keystore_path):
         keystore_identity_ca_key_path,
     )
 
+    # Create new CA if one doesn't already exist
     if not all(os.path.isfile(x) for x in required_files):
-        print('creating new CA key/cert pair')
         _create_ca_key_cert(keystore_ca_key_path, keystore_ca_cert_path)
-        _utilities.create_symlink(src='ca.cert.pem', dst=keystore_permissions_ca_cert_path)
-        _utilities.create_symlink(src='ca.key.pem', dst=keystore_permissions_ca_key_path)
-        _utilities.create_symlink(src='ca.cert.pem', dst=keystore_identity_ca_cert_path)
-        _utilities.create_symlink(src='ca.key.pem', dst=keystore_identity_ca_key_path)
-    else:
-        print('found CA key and cert, not creating new ones!')
-
-    # create governance file
-    gov_path = os.path.join(keystore_path, _KS_ENCLAVES, 'governance.xml')
-    if not os.path.isfile(gov_path):
-        print('creating governance file: %s' % gov_path)
+        _utilities.create_symlink(
+            src=pathlib.Path('ca.cert.pem'), dst=keystore_permissions_ca_cert_path)
+        _utilities.create_symlink(
+            src=pathlib.Path('ca.key.pem'), dst=keystore_permissions_ca_key_path)
+        _utilities.create_symlink(
+            src=pathlib.Path('ca.cert.pem'), dst=keystore_identity_ca_cert_path)
+        _utilities.create_symlink(
+            src=pathlib.Path('ca.key.pem'), dst=keystore_identity_ca_key_path)
+
+    # Create governance file if it doesn't already exist
+    gov_path = keystore_path.joinpath(_KS_ENCLAVES, 'governance.xml')
+    if not gov_path.is_file():
         _create_governance_file(gov_path, _utilities.domain_id())
-    else:
-        print('found governance file, not creating a new one!')
 
-    # sign governance file
-    signed_gov_path = os.path.join(keystore_path, _KS_ENCLAVES, 'governance.p7s')
-    if not os.path.isfile(signed_gov_path):
-        print('creating signed governance file: %s' % signed_gov_path)
+    # Sign governance file if it hasn't already been signed
+    signed_gov_path = keystore_path.joinpath(_KS_ENCLAVES, 'governance.p7s')
+    if not signed_gov_path.is_file():
         _utilities.create_smime_signed_file(
             keystore_permissions_ca_cert_path,
             keystore_permissions_ca_key_path,
             gov_path,
             signed_gov_path)
-    else:
-        print('found signed governance file, not creating a new one!')
-
-    print('all done! enjoy your keystore in %s' % keystore_path)
-    print('cheers!')
-    return True
 
 
-def is_valid_keystore(path):
+def is_valid_keystore(path: pathlib.Path) -> bool:
```

Comment:
Maybe accept path as `Union[str, bytes, os.PathLike]`, then cast it to pathlib.Path? That way users can just shove a string in without issue

Reply:
This applies throughout the public API

Reply:
Feels pretty legacy, particularly when parts of the API return `pathlib.Path`. Not sure I see the benefit unless the API was already public and we wanted to move to pathlib.